src/world/Manager/MapMgr.cpp

```diff
@@ -0,0 +1,889 @@
+
+    for( const auto& group : planeventLgb.groups )
+    {
+      for( const auto& pEntry : group.entries )
+      {
+        if( pEntry->getType() == LgbEntryType::EventNpc )
+        {
+          auto pNpc = reinterpret_cast< LGB_ENPC_ENTRY* >( pEntry.get() );
+
+          auto eNpcData = exdData.get< Data::ENpcBase >( pNpc->data.enpcId )->eNpcData;
+
+          for( auto npcData : eNpcData )
+          {
+            if( npcData == 0 )
+              continue; // Some npcs have data gaps, so we have to iterate through the entire array
+
+            if( count++ % 1000 == 0 )
+              std::cout << ".";
+
+            EventData eventData;
+            eventData.levelId = pNpc->data.instanceId;
+            eventData.actorId = pNpc->data.enpcId;
+
+            auto eventHandlerType = static_cast< EventHandler::EventHandlerType >( npcData >> 16 );
+
+            switch( eventHandlerType )
+            {
+              case EventHandler::EventHandlerType::GuildLeveAssignment:
+              {
+                eventData.iconId = exdData.get< Data::EventIconType >( 5 )->mapIconAvailable + 1;
+
+                m_mapData[ territoryTypeId ].insert( std::make_pair( npcData, eventData ) );
+                break;
+              }
+
+              case EventHandler::EventHandlerType::CustomTalk:
+              {
+                // Include only the beginner arena icon yet. There a few other ones, that aren't referenced in the game files (Some examples are: The Triple Triad Tournament npc which has multiple icons and the ocean fishing icon)
+                if( npcData == 721223 )
+                {
+                  auto customTalk = exdData.get< Data::CustomTalk >( npcData );
+
+                  eventData.iconId = customTalk->iconMap;
+
+                  m_mapData[ territoryTypeId ].insert( std::make_pair( npcData, eventData ) );
+                }
+                break;
+              }
+
+              case EventHandler::EventHandlerType::GuildOrderGuide:
+              {
+                eventData.iconId = exdData.get< Data::EventIconType >( 6 )->mapIconAvailable + 1;
+
+                m_mapData[ territoryTypeId ].insert( std::make_pair( npcData, eventData ) );
+                break;
+              }
+
+              case EventHandler::EventHandlerType::TripleTriad:
+              {
+                eventData.iconId = exdData.get< Data::EventIconType >( 7 )->mapIconAvailable + 1;
+
+                m_mapData[ territoryTypeId ].insert( std::make_pair( npcData, eventData ) );
+                break;
+              }
+
+              case EventHandler::EventHandlerType::PreHandler:
+              {
+                //I think this is used in Bozja and Zadnor, need evidence
+                //m_mapData[ territoryTypeId ].insert( std::make_pair( eventData, 0 ) );
+                break;
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+  std::cout << "\n";
```

Comment:
also use the logger to write out instead of iostream

Reply:
Ok I change that, but I thought this is ok, because in the `InstanceObjectCache` initialization it's done in the same way.

Reply:
more just some debugging code that was left behind lol